### DISCUSSION TITLE
Nested tables for relationship

### DISCUSSION THREAD

**Discussion by ericksonsv** [filamentphp/filament#10667] [Ideas]
Hello, first of all I would like to thank you for the excellent work you have done with FIlament.

I am currently migrating an old project to use FIlament and I would like to know the possibility of adding this new functionality if it does not yet exist to be able to nest several tables with their relationships, the project consists of orders and their service orders, to be able to view the services orders related to each order as you can see in the attached image.

thank you so much

![image](https://github.com/filamentphp/filament/assets/14256724/7f7eeeda-f181-48b1-ac39-9a435f564f2e)

**Comment by helloreece** [comment]
I'm pretty sure this isn't natively supported out the box. However, you might be able to get pretty close with a combination of:

[Collapsible content](https://filamentphp.com/docs/3.x/tables/layout#collapsible-content)

and

[Custom table column](https://filamentphp.com/docs/3.x/tables/columns/custom)

I'd maybe try creating a collapsible column with ->collapsed(false) so that it is always expanded. And then in that collapsed content create a custom column that returns a view where the child table is displayed.

This is a bit hacky, but hopefully helpful. Good luck.

**Reply by ericksonsv** [reply]
Hello helloreece, I had just started the process using collapse and custom row and this is what it looks like, thanks for the recommendation.

![Screenshot from 2024-01-09 23-42-15](https://github.com/filamentphp/filament/assets/14256724/5e400a2c-b630-4347-a739-cad64414391a)

**Reply by albertobenavides** [reply]
Hello. Can you share how you are doing this?

**Reply by davidatperez** [reply]
> Hello. Can you share how you are doing this?

Not the author, but seems like he's grouping the records in the table. I am having a similar approach by adding to :

            ->groups([
                Group::make('parent.id')
                ->label('Order #'),
            ])
            
          Reference: https://filamentphp.com/docs/3.x/tables/grouping#setting-a-grouping-label

**Reply by beuzathor** [reply]
I would also be interested in a more detailed response, particularly regarding the aspect of the edit button within the internal lines of the groups.

**Comment by ericksonsv** [comment]
@beuzathor  @albertobenavides
Hi guys!... sorry for the delay, i going to share the full code, i was using customPage generated for filament and grouping the rows like @davidatperez  said

```
<?php

namespace App\Filament\Resources\OrderResource\Pages;

use App\Filament\Resources\OrderResource;
use App\Models\Order;
use App\Models\OrderService;
use Filament\Forms\Components\TextInput;
use Filament\Forms\Concerns\InteractsWithForms;
use Filament\Forms\Contracts\HasForms;
use Filament\Resources\Pages\Page;
use Filament\Tables\Actions\Action;
use Filament\Tables\Actions\CreateAction;
use Filament\Tables\Actions\EditAction;
use Filament\Tables\Columns\TextColumn;
use Filament\Tables\Concerns\InteractsWithTable;
use Filament\Tables\Contracts\HasTable;
use Filament\Tables\Grouping\Group;
use Filament\Tables\Table;

class ManageOrders extends Page implements HasForms, HasTable
{
    use InteractsWithTable;
    use InteractsWithForms;

    protected static string $resource = OrderResource::class;

    protected static string $view = 'filament.resources.order-resource.pages.manage-orders';

    public function table(Table $table): Table
    {
        return $table
            ->query(OrderService::query())
            ->columns([
                TextColumn::make('id')->sortable()->searchable(),
                TextColumn::make('pickup_date')->sortable()->searchable(),
                TextColumn::make('orderServiceStatus.status')->label('Status')->sortable()->searchable(),
                TextColumn::make('orderServiceType.type')->label('Type')->sortable()->searchable(),
            ])
            ->groups([
                Group::make('order.id')
                    ->label('Order #')
                    ->getDescriptionFromRecordUsing(
                        fn (OrderService $record): string
                        => 'Company: '. $record->order->company->tradename. ' Customer: '. $record->order->customer->name
                    )
                    ->collapsible()
            ])
            ->defaultGroup('order.id')
            ->filters([
                // ...
            ])
            ->actions([
                CreateAction::make()
                    ->model(OrderService::class)
                    ->form([
                        TextInput::make('title')
                            ->required()
                            ->maxLength(255),
                        // ...
                    ]),
                Action::make('edit')
                    // ->url(fn (OrderService $record): string => route('filament.admin.resources.orders.edit', $record->order->id))
                    ->fillForm(fn (OrderService $record): array => [
                        'pickup_date' => $record->pickup_date,
                    ])
                    ->model(OrderService::class)
                    ->form([
                        TextInput::make('pickup_date')
                            ->required()
                            ->maxLength(255),
                        // ...
                    ]),
            ])
            ->bulkActions([
                // ...
            ]);
    }
}

```
